Makefile

```diff
@@ -23,7 +27,7 @@ endif
 
 ifeq ($(COMPILER),gcc)
   CFLAGS += -DCOMPILER_GCC
-  CPPFLAGS += -DCOMPILER_GCC
+  CPPFLAGS += -DCOMPILER_GCC -DAVOID_UB
```

Comment:
I may be wrong but I don't think `CPPFLAGS` are passed to gcc, or that cpp output is passed to gcc, anywhere, so this looks like it would cause `AVOID_UB` to not be defined in gcc builds except for assembly (which uses cpp)

Reply:
`CPPFLAGS` is traditionally meant to be passed to all translation programs that preprocess their input with cpp, e.g. translation programs for C, C++, Fortran, and preprocessed assembler. For example, the default C and C++ recipes for make are `$(CC) $(CPPFLAGS) $(CFLAGS) -c` and `$(CXX) $(CPPFLAGS) $(CXXFLAGS) -c` respectively.

Reply:
that makes sense
(but it's not what oot's makefile does afaict :smile: )